File: problems-extend/1901-2000/1968_array-with-elements-not-equal-to-average-of-neighbors/solution.cpp
#include <algorithm>
#include <vector>

class Solution {
  public:
    vector<int> rearrangeArray(vector<int>& nums) {
        // Sort, then interleave halves: the larger half occupies the even
        // indices, the smaller half the odd ones. Each even-indexed value
        // is then strictly above both (lower-half) neighbors and each
        // odd-indexed value strictly below both (upper-half) neighbors,
        // so no interior element can equal the average of its neighbors.
        sort(nums.begin(), nums.end());
        int n = (int)nums.size();
        vector<int> ans(n);
        for (int k = 0; k < n - n / 2; ++k) {
            ans[2 * k] = nums[n / 2 + k];
        }
        for (int k = 0; k < n / 2; ++k) {
            ans[2 * k + 1] = nums[k];
        }
        return ans;
    }
};

File: problems-extend/1901-2000/1968_array-with-elements-not-equal-to-average-of-neighbors/solution.ts
function rearrangeArray(nums: number[]): number[] {
    // Sort, then interleave halves: the larger half occupies the even
    // indices, the smaller half the odd ones. Each even-indexed value is
    // then strictly above both (lower-half) neighbors and each
    // odd-indexed value strictly below both (upper-half) neighbors, so no
    // interior element can equal the average of its neighbors.
    nums.sort((a, b) => a - b);
    const n = nums.length;
    const ans = new Array(n);
    const half = Math.floor(n / 2);
    for (let k = 0; k < n - half; ++k) ans[2 * k] = nums[half + k];
    for (let k = 0; k < half; ++k) ans[2 * k + 1] = nums[k];
    return ans;
}
